serpentTools/objects/readers.py

```diff
@@ -26,14 +27,38 @@ def __str__(self):
         return '<{} reading {}>'.format(self.__class__.__name__, self.filePath)
 
     def read(self):
+        """The main method for reading that not only parses data, but also
+        runs pre and post checks.
+        """
+        self._precheck()
+        self._read()
+        self._postcheck()
+
+    @abstractmethod
```

Comment:
This is new to me. How is the `abstractmethod` decorator different that having the base `_read` method simply raise `NotImplementedError`? From what I read, this approach does not allow even creating a reader that hasn't overwritten the abstract methods?

Reply:
You're exactly right. I am of the opinion that the code's more expressive this way. The anticipation is that something from BaseReader will always be calling a _read, _precheck, and _postcheck class. If you think some derived classes of the future will not use pre/post check, I'll definitely switch these back to NotImplemented.

Reply:
I'm not sure, but I think this will work well. That way, we have to determine, for each reader, if a pre/post check is applicable. My gut says that post check should always have something, while precheck, 😐, just make sure this is the right type of file? Like you have written with the branching reader, ensure that we can extract the total number of coefficients from the first line.